pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -176,7 +175,14 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pr *v1beta1.PipelineRun)
 		// and may not have had all of the assumed default specified.
 		pr.SetDefaults(contexts.WithUpgradeViaDefaulting(ctx))
 
-		c.updatePipelineResults(ctx, pr, getPipelineFunc)
+		if _, pipelineSpec, err := resources.GetPipelineData(ctx, pr, getPipelineFunc); err != nil {
+			msg := fmt.Sprintf("Failed to get Pipeline Spec to process Pipeline Results for PipelineRun %s/%s: %v", pr.Namespace, pr.Name, err)
+			logger.Error(msg)
+			pr.Status.MarkFailed(ReasonCouldntGetPipeline, msg)
```

Comment:
I know this has been discussed already, but I think we should not change the final condition of a pipeline once it has been already set since it would break the meaningfulness of the condition. Consumers would not have a way anymore to know if a pipeline status is final or not.
We could to move this result logic so that we evaluate it before the condition is set; if you prefer not to do that in this PR, we could for now log / emit a warning.

Reply:
I'll add a log line / warning temporarily as this PR is already large enough that I'd prefer not to bloat it further (and given this is the current state of reconciler in `master` I feel it's safer to make the change separately).  Created https://github.com/tektoncd/pipeline/issues/3667 to tackle this immediately afterwards.

Reply:
Thanks for reviewing @afrittoli .  I'm now logging a warning that the existing PipelineRun Succeeded condition is going to be overwritten.  As part of that message I've recorded which PipelineRun is affected and the existing Succeeded condition before this new failure is introduced.

Reply:
Thanks for the update @sbwsg. It seems to me that updating a pipeline success status after completion would be introducing an issue, so this patch would fix an issue in favour of introducing another one. 
I understand and agree that you don't want to bloat this PR, but I think a better alternative here would be to send an event warning the user that while the pipeline succeeded the results are not reliable as there was a failure.
Alternatively I would recommend waiting for the next release, so that this PR and the follow-up may be released together.

Reply:
> It seems to me that updating a pipeline success status after completion would be introducing an issue, so this patch would fix an issue in favour of introducing another one.

@afrittoli I just want to clarify here.  Our existing code in master already overwrites the pipeline status if it fails to get the pipeline data while processing Pipeline Results.  You can see this happening here: https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/pipelinerun/pipelinerun.go#L320-L333

That function, `updatePipelineResults`, is called from https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/pipelinerun/pipelinerun.go#L179 which is inside a block guarded by `if pr.IsDone()`.  This definitely isn't a new issue I'm introducing in order to fix my other issue I promise 😅

Reply:
Thanks for the clarification, I didn't realise that 🙏